test/ToucanRegenBridge.js

```diff
@@ -144,7 +144,7 @@ describe("Bridge contract", function () {
 	describe("Regen to Polygon", function () {
 		it("should fail with non positive amount", async function () {
 			await expect(
-				bridge.connect(broker).issueTCO2Tokens(regenUser, broker.address, tco2.address, 0, "test")
+				bridge.connect(tokenIssuer).issueTCO2Tokens(regenUser, broker.address, tco2.address, 0, "test")
```

Comment:
Good catch but how did this pass previously?

Reply:
That is an excellent question that I couldn't answer myself. The error was, for whatever reason, introduced in https://github.com/regen-network/toucan-bridge/pull/61/commits/2d63ffef795a3ac197bdb6d500f5e547606ec247 and wasn't an issue previously.

But it's bizarre because there is no specific place where a `broker` becomes a token issuer before that, not in the contract or test. The flow previously is: the contract gets deployed with `bridgeAdmin` (now `tokenIssuer`) as a token issuer. At one [later point](https://github.com/regen-network/toucan-bridge/blob/master/test/ToucanRegenBridge.js#L326), the token issuer role is given to `admin`. But never to `broker`.

Reply:
I agree this seems odd, and thanks for the analysis.

2d63ffe is no longer part of this PR; however I see that 22536b8f (which *is* in the PR) is essentially identical, except for doing `require(hasRole(...))` instead of `onlyRole(...)`.

Does 22536b8f introduce the same error?  If not, then you could potentially compare them to figure out where the error crept in, although maybe it's not worth the effort given the urgency to get other stuff done.

However if it does then this means an error was introduced in a previous commit in this PR, and then fixed here in this commit - if so, then as a reminder that would be going against part of [our coding standards](https://github.com/ToucanProtocol/tokenizer/blob/master/documentation/coding-standards.md):

![image](https://user-images.githubusercontent.com/100738/221428278-ee90d113-f4f2-4e32-a51a-75bb9c3522ee.png)

It's totally fine if we can't fully explain why the breakage happened, as long as we explicitly call that out and make sure the above golden rule wasn't broken.  If it was (i.e. if 22536b8f introduced the same error as 2d63ffe), then the simple solution would be to squash these two commits together, and explain the uncertainty in the commit message.  The information you gave in the comment above is already very helpful, so you could just include that in the combined commit message.

Reply:
Agree about squashing them. I do feel that keeping them separate made the review easier.